src/server/frontend/wayland/wl_surface.h

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright © 2018 Canonical Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Authored by: Christopher James Halse Rogers <christopher.halse.rogers@canonical.com>
+ *              William Wold <william.wold@canonical.com>
+ */
+
+#ifndef WL_SURFACE_H
+#define WL_SURFACE_H
+
+#include "core_generated_interfaces.h"
+
+#include "mir/frontend/buffer_stream_id.h"
+#include "mir/frontend/surface_id.h"
+
+#include <vector>
+
+namespace mir
+{
+class Executor;
+
+namespace graphics
+{
+class WaylandAllocator;
+}
+
+namespace frontend
+{
+class BufferStream;
+struct WlMirWindow;
+
+class WlSurface : public wayland::Surface
+{
+public:
+    WlSurface(wl_client* client,
+              wl_resource* parent,
+              uint32_t id,
+              const std::shared_ptr< mir::Executor >& executor,
+              const std::shared_ptr< mir::graphics::WaylandAllocator >& allocator);
```

Comment:
Coding style - should be `std::shared_ptr<graphics::WaylandAllocator> const&` etc.